File: dd-java-agent/benchmark-integration/jetty-perftest/src/main/java/datadog/perftest/Worker.java
package datadog.perftest;

import static datadog.trace.instrumentation.api.AgentTracer.activeSpan;

import datadog.trace.api.Trace;
import datadog.trace.instrumentation.api.AgentSpan;
import java.util.concurrent.TimeUnit;

public class Worker {

  @Trace
  /** Simulate work for the give number of milliseconds. */
  public static void doWork(final long workTimeMS) {
    final AgentSpan span = activeSpan();
    if (span != null) {
      span.setTag("work-time", workTimeMS);
      span.setTag("info", "interesting stuff");
      span.setTag("additionalInfo", "interesting stuff");
    }

    final long doneTimestamp = System.nanoTime() + TimeUnit.MILLISECONDS.toNanos(workTimeMS);
    while (System.nanoTime() < doneTimestamp) {
      // busy-wait to simulate work
    }
  }
}

File: dd-java-agent/benchmark-integration/play-perftest/app/controllers/Worker.scala
package controllers

import datadog.trace.instrumentation.api.AgentTracer.activeSpan

import datadog.trace.api.Trace
import java.util.concurrent.TimeUnit

object Worker {
  @Trace
  def doWork(workTimeMS: Long) = {
    val span = activeSpan
    if (span != null) {
      span.setTag("work-time", workTimeMS)
      span.setTag("info", "interesting stuff")
      span.setTag("additionalInfo", "interesting stuff")
    }
    val doneTimestamp = System.nanoTime + TimeUnit.MILLISECONDS.toNanos(workTimeMS)
    while ( {
      System.nanoTime < doneTimestamp
    }) {
      // busy-wait to simulate work
    }
  }
}
